src/osd/ReplicatedBackend.cc

```diff
@@ -2180,8 +2234,9 @@ void ReplicatedBackend::trim_pushed_data(
   }
 }
 
-void ReplicatedBackend::_failed_push(pg_shard_t from, const hobject_t &soid)
+void ReplicatedBackend::_failed_pull(pg_shard_t from, const hobject_t &soid)
 {
+  dout(20) << __func__ << ": " << soid << " from " << from << dendl;
   list<pg_shard_t> fl = { from };
   get_parent()->failed_push(fl, soid);
```

Comment:
Not sure what you mean here.  What should failed_push() be called?  Sometimes failed_push() is called when errors during push and it is called from _failed_pull() when a pull failed to read at remote OSD.